app/javascript/packages/document-capture/components/submission.jsx

```diff
@@ -6,6 +6,40 @@ import SubmissionComplete from './submission-complete';
 import SubmissionInterstitial from './submission-interstitial';
 import CallbackOnMount from './callback-on-mount';
 
+/**
+ * Given an object where some values may be promises, returns a promise which resolves to an object
+ * with the same keys and all resolved promise values.
+ *
+ * @template {Record<string, any>} T
+ *
+ * @param {T} object
+ *
+ * @return {Promise<Record<keyof T, any>>} Object with all values resolved.
+ */
+export async function resolveObjectValues(object) {
+  const resolved = {};
+
+  for (const [key, value] of Object.entries(object)) {
+    // Disable reason: While typically inadvisable since await blocks continued iteration of the
+    // loop, the intent of the function is to not resolve until all member values have settled.
+    // eslint-disable-next-line no-await-in-loop
+    resolved[key] = await value;
+  }
+
+  return /** @type {Record<keyof T, any>} */ (resolved);
+}
+
+/**
+ * Returns a function which runs an array of promise creator functions in series (sequential) order.
+ *
+ * @param {Array<(...args:any)=>Promise<any>>} promiseCreators Promise creator functions.
+ *
+ * @return {(...args:any)=>Promise<any>} Promise resolving once all promise creators in series have
+ * run.
+ */
+export const series = (promiseCreators) => (value) =>
```

Comment:
WDYT of making this a variadic function? Then we don't have to manually create an array:

```
(promiseCreators....) => (value) => 
series(promiseA, promiseB)
```

Reply:
> WDYT of making this a variadic function? Then we don't have to manually create an array:

That seems fine. I think I'd seen it as something akin to functions like `Promise.all`, `Promise.race`, etc, which accept their arguments as an array, but the usage isn't quite the same (function composition).

I'll update 👍

Reply:
> WDYT of making this a variadic function? Then we don't have to manually create an array:

Updated in d8b681f42.